newIDE/app/src/MainFrame/index.js

```diff
@@ -1238,6 +1245,18 @@ const MainFrame = (props: Props) => {
     toolbar.current.setEditorToolbar(editorToolbar);
   };
 
+  const hideEditorToolbar = (hidden: boolean) => {
```

Comment:
Could this has been done declaratively rather than imperatively? I.e: passing a props rather than using a ref?

Reply:
Let's useCallback this because otherwise this changes at each render and will trigger the effects inside child component at every render. :)

Reply:
Indeed, setEditorToolbar (just above) does not have a useCallback and is passed to all the editors, we should probably do the same for it?